compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -77,11 +77,12 @@ where
 
     /// Add a subdiagnostic to an existing diagnostic where `f` is invoked on every message used
     /// (to optionally perform eager translation).
-    fn add_to_diagnostic_with<F>(self, diag: &mut Diagnostic, f: F)
-    where
-        F: Fn(&mut Diagnostic, SubdiagnosticMessage) -> SubdiagnosticMessage;
+    fn add_to_diagnostic_with<F: SubdiagnosticMessageOp>(self, diag: &mut Diagnostic, f: F);
 }
 
+pub trait SubdiagnosticMessageOp =
```

Comment:
Note: This could also just be expressed as:

```rust
trait SubdiagnosticMessageOp: Fn(&mut Diagnostic, SubdiagnosticMessage) -> SubdiagnosticMessage {}
impl<F> SubdiagnosticMessageOp for F where F: Fn(&mut Diagnostic, SubdiagnosticMessage) -> SubdiagnosticMessage {}
```

SO I don't believe this commits us to any particularities of the `trait_alias` feature yet.

Reply:
There's one other use of `trait_alias` in the compiler so I figured it was fair game. I think this comment is expressing the opinion that using `trait alias` is ok?

Reply:
Yes, it's ok